### PR TITLE
Upgrade Jekyll, Docs Build Process and Add Docs Contributing Guide

### PR DESCRIPTION
**Pull request by a93h**
* upgraded Jekyll version and gems
* uses NodeJS instead of therubyracer for v8 engine
* stored old gems versions in Gemfile.lock.backup
* added new guide in docs on how to contribute to docs

### CODE REVIEW

**Review by davegson — COMMENTED** [review]
hey there, thanks for your PR! Finally got the time to glance over it. From what I can tell this looks good. I do not think we need the backup file, since the git technology does this well. So you can just delete that.

I will hopefully get the chance to look ever the texts of the guides next week - but liking the idea behind it 👍

**Comment by a93h** [comment]
@davegson Thanks very much, the backup file has been removed.

I have some unique use cases for Portmaster, yet haven't had the time to invest in developing my use case.

Once I have the time, I will though.

**Review by davegson — APPROVED** [review]

File: _data/menus.yml [hunk 1]
```diff
@@ -7,6 +7,9 @@ items:
         extra_path: "/contribute"
         children:
           - title: Contribute
+            children:
+              - title: Docs Guide
+                extra_path: "/docs-guide"
```
Comment (davegson) [inline _data/menus.yml:12]: as an fyi, you do not need the `extra_path` in this case, as it automatically creates the path "docs-guide". `extra_path` is just needed for creating the "previous path" for child elements (sounds confusing, happy to explain further if it's unclear)

anyway, I'll quickly remove this on the `master` branch

**Comment by davegson** [comment]
thanks for the PR! I went ahead and merged it

**Pull request merged**